cmd/query/app/apiv3/http_gateway.go

```diff
@@ -72,11 +71,11 @@ func (h *HTTPGateway) addRoute(
 	if h.TenancyMgr.Enabled {
 		handler = tenancy.ExtractTenantHTTPHandler(h.TenancyMgr, handler)
 	}
-	traceMiddleware := otelhttp.NewHandler(
-		otelhttp.WithRouteTag(route, handler),
-		route,
-		otelhttp.WithTracerProvider(h.Tracer))
-	return router.HandleFunc(route, traceMiddleware.ServeHTTP)
+	// traceMiddleware := otelhttp.NewHandler(
+	// 	otelhttp.WithRouteTag(route, handler),
+	// 	route,
+	// 	otelhttp.WithTracerProvider(h.Tracer))
```

Comment:
@yurishkuro i tried commenting out this middleware to see if the GRPC tests would pass but it looks like the grpc v2 tests are running into the same issue of getting stuck in a loop of `WriteSpan`. Any idea why that's happening and if there's a way around it?

Reply:
why are grpc tests affected by http changes? We already merged grpc PR on green CI.

Reply:
its not the grpc server tests - its the grpc storage tests (https://github.com/jaegertracing/jaeger/actions/runs/11526027865/job/32089447738?pr=6121)